lib/request_handler/validation/dry_engine.rb

```diff
@@ -7,34 +7,45 @@ module RequestHandler
   module Validation
     class DryEngine < Engine
       def self.valid_schema?(schema)
-        schema.respond_to?(:call)
+        schema_instance?(schema) || schema_class?(schema)
       end
 
       def self.validate(value, schema, options: {})
-        value = value.deep_symbolize_keys if schema.respond_to?(:rules) &&
-                                             schema.rules.keys.first.is_a?(Symbol)
-        result = if options.empty?
-                   schema.call(value)
-                 else
-                   schema.with(options).call(value)
-                 end
-
-        to_result(result)
+        value = value.deep_symbolize_keys if value.is_a?(Hash)
+        to_result(call_schema(value, schema, options))
       rescue Dry::Types::ConstraintError => e
         Result.new(output: nil, errors: { '' => e })
       end
 
+      def self.call_schema(value, schema, options)
+        if options.empty?
+          schema_instance?(schema) ? schema.call(value) : schema.new.call(value)
+        else
+          schema_instance?(schema) ? schema.with(options).call(value) : schema.new(options).call(value)
+        end
+      end
+
+      def self.schema_instance?(schema)
+        schema.respond_to?(:call)
```

Comment:
hmm is it the best to look for an instance just by relying on it having a `call` method vs a `schema` method? not sure that is robust

Reply:
I'm not sure. I didn't know how else to distinguish instances from classes. Do you have an idea?